### PR TITLE
Move build package to it's own directory

### PR DESCRIPTION
**Pull request by natebosch**
Rather than nest build_test and the end-to-end examples keep build as
it's own directory. When this is packaged up with pub these directories
are currently coming with but are not usable.

This is the first step for issue #140

### CODE REVIEW

**Comment by natebosch** [comment]
Most of the changes aren't uninteresting file moves.

The ones to pay attention to are:
new top level README.md
changes in pubspec.yaml
change in travis.sh

File: tool/travis.sh [hunk 1]
```diff
@@ -7,13 +7,14 @@
 # Fast fail the script on failures.
 set -e
 
-# Verify that the libraries are error free, requires some strong mode changes
-# that are only in the dev channel right now though.
-if [ "$TRAVIS_DART_VERSION" != "stable" ]; then
-  dartanalyzer --fatal-warnings lib/build.dart build_test/lib/build_test.dart
-fi
+cd build_test
```
Comment (jakemac53) [inline tool/travis.sh]: nit: maybe use pushd/popd?

File: README.md [hunk 2]
```diff
@@ -1,282 +1,23 @@
 # [![Build Status](https://travis-ci.org/dart-lang/build.svg?branch=master)](https://travis-ci.org/dart-lang/build) [![Coverage Status](https://coveralls.io/repos/dart-lang/build/badge.svg?branch=master)](https://coveralls.io/r/dart-lang/build)
 
-This package provides a way of generating files using Dart code, outside of
-`pub`. These files are generated directly on disk in the current package folder,
-and rebuilds are _incremental_.
+These packages provide utilities for generating Dart code outside of `pub`.
```
Comment (jakemac53) [inline README.md]: for the general readme, lets remove `outside of pub`, as you will be able to use the barback package to run builders using pub.

File: build/README.md [hunk 3]
```diff
@@ -0,0 +1,282 @@
+# [![Build Status](https://travis-ci.org/dart-lang/build.svg?branch=master)](https://travis-ci.org/dart-lang/build) [![Coverage Status](https://coveralls.io/repos/dart-lang/build/badge.svg?branch=master)](https://coveralls.io/r/dart-lang/build)
```
Comment (jakemac53) [inline build/README.md]: We could add this into the main readme as well?

File: test/fixtures/basic_pkg/.packages [hunk 4]
```diff
@@ -1,6 +0,0 @@
-# Generated by hand on 2016-02-01
```
Comment (jakemac53) [inline test/fixtures/basic_pkg/.packages]: These should probably be moves not deletes, maybe something in the gitignore?

File: build/README.md (hunk 3 above)
Comment (natebosch) [inline build/README.md]: It's already there...

File: test/fixtures/basic_pkg/.packages (hunk 4 above)
Comment (natebosch) [inline test/fixtures/basic_pkg/.packages]: oops, fixed

File: tool/travis.sh (hunk 1 above)
Comment (natebosch) [inline tool/travis.sh]: done

File: build/README.md (hunk 3 above)
Comment (jakemac53) [inline build/README.md]: lol, sweet then!

File: tool/travis.sh [hunk 5]
```diff
@@ -26,6 +28,3 @@ if [ "$COVERALLS_TOKEN" ] && [ "$TRAVIS_DART_VERSION" = "stable" ]; then
     tool/test_all.dart
   rm tool/test_all.dart
 fi
```
Comment (jakemac53) [inline tool/travis.sh]: maybe add a popd here just so we can add things later without confusing errors

Comment (natebosch) [inline tool/travis.sh]: Good call. Done

**Pull request merged**